### ISSUE TITLE
ctrld feature requests

### ISSUE THREAD

**Issue by pmcarrion** [Control-D-Inc/ctrld#40]
Hello,

I have some feature requests for ctrld:

1. Auto-update ctrld to the latest version.

Some days ago, you released version 1.1.4, and I installed on my computer.

However, I just checked the version and it showed this:
```
sudo ./ctrld --version
Password:
ctrld version v1.1.5-next-9c22701
```
Have you implemented an auto-update feature in ctrld?

2. Use macOS notifications to alert users of several ctrld service events:

- status / started / stopped / restarted / uninstalled
- DNS set / not set
- interface used / changed
- auto-updated binary / config.toml
- etc.

3. Include ctrld in [Homebrew](https://brew.sh), so it can be installed and updated through that package manager.

4. Utility for Asus-Merlin and Tomato routers as mentioned [here](https://docs.controld.com/docs/device-clients).

5. Self DNS-resolving speedtest in ctrld, and choose other protocols (Do3, DoQ or DoT) if they are faster than DoH for the current network.

6. Automatically set up the DNS server if there's a new network interface active.

Thanks.

**Comment by cuonglm** [comment]
> Have you implemented an auto-update feature in ctrld?

Nope, that probably a tagging problem of go releaser, we will check this.

**Comment by cuonglm** [comment]
> > Have you implemented an auto-update feature in ctrld?
> 
> Nope, that probably a tagging problem of go releaser, we will check this.

@pmcarrion We updated MacOS binaries to correct the tag!

PS: There's no different in term of features between old and new binary (both are built at commit 9c22701), just the version generated is different when building using goreleaser.

**Comment by pmcarrion** [comment]
> > > Have you implemented an auto-update feature in ctrld?
> > 
> > 
> > Nope, that probably a tagging problem of go releaser, we will check this.
> 
> @pmcarrion We updated MacOS binaries to correct the tag!
> 
> PS: There's no different in term of features between old and new binary (both are built at commit [9c22701](https://github.com/Control-D-Inc/ctrld/commit/9c22701940ab3c0865cfb6d33c0342245d92df6e)), just the version generated is different when building using goreleaser.

Thanks, @cuonglm!

```
sudo ./ctrld --version
ctrld version v1.1.4-9c22701
```

Please don’t forget to check the other requests. I think they would be very useful.
